### PR TITLE
feat(metrics): add observable instruments to DiagnosticsMetrics, with trackCache

### PR DESCRIPTION
**Pull request by vsseixaso**
Implements the approved RFC: [vtex/stable-storefronts-specs#8](https://github.com/vtex/stable-storefronts-specs/pull/8), with one decision revisited — see [Departure from the RFC](#departure-from-the-rfc-corte-seco) below.

#### What is the purpose of this pull request?

Adds three public methods to `DiagnosticsMetrics`:

- **`registerObservableGauge(name, observe, options?)`** / **`registerObservableCounter(name, observe, options?)`** — generic pull-based instruments, for values read on a schedule rather than pushed per request (queue depth, pool size, etc). Re-registering a name replaces the previous callback on the same instrument, matching the precedent in [vtex/faststore-cloud's `lib/diagnostics.ts`](https://github.com/vtex/faststore-cloud/blob/main/lib/diagnostics.ts#L659-L670).
- **`trackCache(name, cacheInstance)`** — the direct replacement for the legacy `MetricsAccumulator.trackCache()`. Accepts the same `LRUCache`/`DiskCache`/`LRUDiskCache`/`MultilayeredCache` instances already in use. Emits `io_app_cache_operations_total`, `io_app_cache_items_current`, `io_app_cache_capacity`, `io_app_cache_disposed_total`.

Both reach the OTel SDK through `metricsClient.getProvider()` — already part of `DiagnosticsMetrics`' declared type, and the same access node-vtex-api already uses for `HostMetricsInstrumentation` in `service/telemetry/client.ts`. No new dependency. The meter name matches the one the existing push instruments already use, so the scope stays consistent.

And adds **`getCumulativeStats()`** to the four cache classes — see below for why.

#### What problem is this solving?

`DiagnosticsMetrics` only exposes push-based instruments (`recordLatency`, `incrementCounter`, `setGauge`). `metrics.trackCache()` and `metrics.addOnFlushMetric()` are pull-based and have had no replacement, so apps migrating off `MetricsAccumulator` keep both calls untouched (see [vtex/render-server#861](https://github.com/vtex/render-server/pull/861)) — which blocks ever removing the legacy metrics API from an app that uses either.

#### Departure from the RFC: corte seco

The RFC's open question was what to do about the cache classes' `getStats()` **consuming the counters it reads** — a legacy of `MetricsAccumulator` publishing metrics as log lines in windows. Its approved answer was *corte seco*: don't touch the cache classes, and accept that reading a cache from two places splits its counts.

**This PR does not do that**, and the reason got sharper once #676 came into view.

[#676](https://github.com/vtex/node-vtex-api/pull/676) (merged 2026-08-05, released in 7.4.2) removed the status log write. `statusTrack()` still runs and **still calls `getStats()` on every registered cache, resetting its counters** — but the return value is now discarded. The comment it left behind:

```js
// Flushing resets the metric accumulators, the CPU usage baseline and the
// incoming request stats, so it must keep running even though nothing
// consumes the returned metrics anymore.
```

So under corte seco, an app that registered a cache in both places would leak an arbitrary fraction of its counts into a reader that throws them away — **with no second metric anywhere to reveal the discrepancy**, because the legacy output reaches no backend. Not "each sees about half", as the RFC put it: one plausible-looking, permanently wrong number.

That is the argument. Two consequences:

1. **"Replace, don't add" is unverifiable and its failure is invisible.** It's enforced by a doc paragraph, across 23 call sites in three apps, migrated by different people in different PRs.
2. **A partial or reverted migration stops being safe.** With a side-effect-free read, a half-migrated app reports correctly.

For the same reason, this PR does *not* claim you can validate the new metric against the legacy one — there is nothing to compare against. What you get is that leaving the legacy call in place is harmless.

What's here instead: the four cache classes' internal counters are now plain **monotonic totals**, and `getStats()` computes its own delta against a snapshot of what it last reported. **Its external contract is unchanged** — same windowed values, same `hitRate`, pinned by tests. `getCumulativeStats()` reports the process-lifetime total and has no side effects, so the observable reader and the legacy flush read the same cache independently.

Cost: `+85/-56` across 6 files — but **`-11` net in `DiagnosticsMetrics.ts`**, because the running-total bookkeeping that existed only to undo the reset (`cacheCumulative` and the whole delta→cumulative folding block) is gone. Migrating a cache is now additive:

```typescript
metrics.trackCache('pages', pagesCacheStorage)                      // harmless to keep
global.diagnosticsMetrics?.trackCache('pages', pagesCacheStorage)   // add; drop the line above when convenient
```

A side effect worth naming: since #676 the legacy cache metric reaches nothing, so this PR **restores** cache observability rather than preserving it. The RFC's problem statement assumed the former; corrected in its v1.3.
This also gives the four cache classes their **first tests** — they had none.

`hitRate` is intentionally not republished — derivable from `io_app_cache_operations_total`, and publishing a pre-computed ratio would prevent correct aggregation across instances. A Grafana search across ~1,565 dashboards and ~1,769 alert rules (done while writing the RFC) found no consumer of `hitRate`/`itemCount`/`disposedItems` to preserve.

#### Also in here

**Pending registrations.** `DiagnosticsMetrics`' underlying client initializes asynchronously, but apps call `trackCache`/`registerObservableGauge`/`registerObservableCounter` synchronously at module load — frequently before that initialization finishes. A call made before the client is ready is now queued and replayed once it is. This is the **one existing method this PR touches** (`initMetricClient`), and it's a no-op for any app that never calls the new APIs — the existing tests' mock client has no `getProvider()` at all, and it is never called by them.

**Three guards on the observable instruments:**

- A name already registered as the other kind is **refused with an error log**. The SDK accepts two same-named streams of different types silently (verified: `dup_metric:GAUGE` + `dup_metric:SUM`, no error) and the collector then rejects them.
- Attributes reported by an observable callback are held to the same `MAX_CUSTOM_ATTRIBUTES` limit as the push-based methods. The callback calls `result.observe` directly, so this needs a wrapper on the result.
- The operations counter is skipped for an object with no hit/total counters, instead of emitting two zero-valued series.

#### How should this be manually tested?

```bash
yarn test src/metrics/DiagnosticsMetrics.test.ts src/caches/cacheStats.test.ts
```

The cache tests run against real cache instances; `trackCache`'s tests run against a real `MeterProvider` + `MetricReader` rather than a mock, specifically to catch the single-read-per-cycle and cumulative-counter accounting a mock could get "passing" while still wrong. Notable cases:

- **`delta` temporality** — the one actually configured in `service/telemetry/client.ts`, not just `cumulative`.
- **A real `LRUCache` read by the legacy flush and the observable at once**, confirming neither steals the other's counts.
- **`trackCache` called before the client is ready.**
- **The inert-if-unused guarantee**, spied before construction so it covers initialization too.

Verified:

| | master | this branch |
|---|---|---|
| `tsc --noEmit` | 0 errors | **0 errors** |
| `tslint` | 0 errors | **0 errors** |
| full repo suite | 240 passed / 24 skipped | **277 passed / 24 skipped** |
| coverage, `DiagnosticsMetrics.ts` | — | **97.4% lines** |
| coverage, the 4 cache classes | **0% (no tests existed)** | LRUCache 100%, others 71–82% |

> ⚠️ **The SonarQube quality gate is red, and it is measuring the whole repository, not this PR.** The scan log says why:
>
> ```
> WARN: Shallow clone detected, no blame information will be provided.
>       You can convert to non-shallow with 'git fetch --unshallow'.
> INFO: SCM Publisher 0/197 source files have been analyzed
> WARN: Missing blame information for the following files: [~197 files]
> ```
>
> With no blame data the scanner cannot tell which lines changed, so it counts everything as new code: **`new_lines` = 13,975**, against 18,195 lines in `src/`. Hence 324 issues across 86 files, including ones this PR never touched (`clients/infra/Apps.ts`, `utils/MineWinsConflictsResolver.ts`). The 35.1% "coverage on new code" is likewise the whole repo's coverage — the lcov report is read correctly (`INFO: Analysing [.../coverage/lcov.info]`).
>
> Fixing it means an unshallow fetch in the `node-ci-v2` clone step, which is central DK CI config, not this repo — there is no `sonar-project.properties` on `master` either. It affects every PR on this pipeline.
>
> Of the 40 issues Sonar reports in the 8 files this PR touches, **3 were on lines this PR added** — all `S2933` (never-reassigned member should be `readonly`), fixed. Both new test files report zero. The three `HIGH` findings are pre-existing on `master`: `S4123` at `MultilayeredCache.ts:39,44`, and `S7059` at `DiagnosticsMetrics.ts:193` (`this.initMetricClient()` in the constructor) — that last one is a fair observation, but out of scope here.

#### Types of changes

* [x] New feature (a non-breaking change which adds functionality)
* [x] Requires change to documentation, which has been updated accordingly.

`docs/METRICS_CATALOG.md` gains a **Cache Metrics (Observable)** section, including the caveat that `io_app_cache_capacity` is in the cache's own units (`max` is an item count only when the LRU wasn't built with a `length` function).

`docs/METRICS_OVERVIEW.md` gains **Pattern 5** (`trackCache`) and **Pattern 6** (`addOnFlushMetric` → `registerObservableGauge`/`Counter`) — the RFC names *both* legacy APIs as blocking STR-1209, and the second had no documented path. Cross-referenced with the existing Pattern 4, the different already-supported push-based idiom.

#### Follow-up, deliberately not here

`src/HttpClient/middlewares/inflight.ts` uses `addOnFlushMetric` for node-vtex-api's own inflight-map metric, so the legacy method can't be deleted from `MetricsAccumulator` until that caller moves to `registerObservableGauge`. It also has a latent bug — `inflight.entries.length` is the method's arity (`0`), not the map size, so that metric has always reported zero. Separate PR.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: src/metrics/DiagnosticsMetrics.ts [hunk 1]
```diff
@@ -383,5 +466,299 @@ export class DiagnosticsMetrics {
...
+    }
+
+    const { operations, items, capacity, disposed } = this.cacheInstruments
+
+    for (const [name, cache] of this.cacheRegistry) {
+      let stats: { [key: string]: number | boolean | string | undefined }
+      try {
+        stats = cache.getStats()
+      } catch (error) {
+        console.error(`DiagnosticsMetrics: failed to read stats for cache '${name}':`, error)
```
[the remark this answers is not in the document]
Comment (vsseixaso) [reply src/metrics/DiagnosticsMetrics.ts]: Fixed in 5d8314f — passed `name` as a separate argument to `console.error` instead of interpolating it into the format string.
